core/controllers/base_test.py

```diff
@@ -1423,3 +1469,397 @@ def test_that_correct_derived_class_does_not_raise_exception(self):
         with self.swap(self, 'testapp', self.mock_testapp):
             self.post_json(
                 '/correctmock', payload, expected_status_int=200)
+
+
+class SchemaValidationIntegrationTests(test_utils.GenericTestBase):
+    """Tests all the functionality of SVS(Schema-Validation-System)
+    architecture.
+    """
+    handler_class_names_with_no_schema = (
+        payload_validator.HANDLER_CLASS_NAMES_WITH_NO_SCHEMA)
+    wiki_page_link = (
+        'https://github.com/oppia/oppia/wiki/Validation-of-handler-args')
+
+    def _get_list_of_routes_which_need_schemas(self):
+        """This method iterates over all the routes and returns those routes
+        which need schema.
+
+        Returns:
+            list_of_routes_which_needs_schema: list(RedirectRoute). A list of
+            RedirectRoute objects.
+        """
+        list_of_routes_which_needs_schema = []
+        for route in main.URLS:
+            # TODO(#13139): Remove below check once all the MAPREDUCE_HANDLERS
+            # are removed from the codebase.
+            if isinstance(route, tuple):
+                continue
+            list_of_routes_which_needs_schema.append(route)
+        return list_of_routes_which_needs_schema
+
+    def test_every_handler_class_has_schema(self):
+        """This test ensures that every child class of BaseHandler
+        has an associated schema.
+        """
+        list_of_handlers_which_need_schema = []
+        list_of_routes_which_needs_schema = (
+            self._get_list_of_routes_which_need_schemas())
+
+        for route in list_of_routes_which_needs_schema:
+            handler = route.handler
+
+            handler_class_name = handler.__name__
+            if handler_class_name in self.handler_class_names_with_no_schema:
+                continue
+
+            schema_written_for_request_methods = (
+                handler.HANDLER_ARGS_SCHEMAS is not None)
+            schema_written_for_url_path_args = (
+                handler.URL_PATH_ARGS_SCHEMAS is not None)
+            handler_has_schemas = (schema_written_for_request_methods and
+                schema_written_for_url_path_args)
+
+            if handler_has_schemas is False:
+                list_of_handlers_which_need_schema.append(handler_class_name)
+
+        error_msg = (
+            'Schema required in handlers: [ %s ].'
+            '\nVisit %s to know, how to write schema for handler args.' % (
```

Comment:
to know, how --> to learn how
schema --> schemas

These doc changes should be made elsewhere below, too.

Reply:
Done